src/messages/Modal.ts

```diff
@@ -0,0 +1,18 @@
+import { constant, Decoder, mixed } from 'decoders';
+import { labeledDecoder, LabeledMsg } from './LabeledMsg';
+
+/**
+ * A message used to publish a generic messages
+ * between the clients and the host application.
+ */
+export interface LabeledModal extends LabeledMsg<'modalRequest', any> {
+  /** Message identifier */
+  msgType: 'modalRequest';
+}
+
+const decoder: Decoder<LabeledModal> = labeledDecoder(
+  constant<'modalRequest'>('modalRequest'),
+  mixed
```

Comment:
This should be something that validates that the data is an actual modalRequest so we don't accidentally pass bad data to a host application.  Probably just validating that the request's `modalId` field is a string.

Reply:
Changed as requested.